docs/custom_build.md

```diff
@@ -0,0 +1,55 @@
+---
+title: Non-Docker Builds
+layout: docs
+---
+
+`docker build` is the common way to build container images, but there are others. [Bazel](https://github.com/google/bazel), [s2i](https://github.com/openshift/source-to-image), and [ko](https://github.com/google/ko) support building images. Tilt supports these other tools with the function [`custom_build`](https://docs.tilt.dev/api.html#api.custom_build) instead of `docker_build`.
+
+## Usage
+Building an image with a custom build tool requires:
+* Name of the image to build (as a ref, e.g. `frontend` or `gcr.io/company-name/frontend')
+* Command to run (e.g. `bazel build //frontend:image` or `build_frontend.sh`)
+* Files and directories that are dependencies (e.g. `['frontend']` or `['frontend', 'util', 'data.txt']`)
```

Comment:
I might add something like "(this is used to determine when to kick off builds)".
explicitly specifying dependencies can be annoying bookkeeping, and it's useful to establish an understanding of what happens when the bookkeeping fails, to get an idea of the risk involved, how meticulous one needs to be about the bookkeeping, and be more likely to quickly recognize when the bookkeeping hasn't happened.